library/src/cache/patch_manager.rs

```diff
@@ -0,0 +1,883 @@
+                number: patch_number,
+                size: patch_file_contents.len() as u64
+            })
+        );
+        assert!(!file_path.exists());
+        assert_eq!(manager.highest_seen_patch_number(), Some(patch_number));
+    }
+
+    #[test]
+    fn does_not_set_higher_highest_seen_patch_number_if_added_patch_is_lower() -> Result<()> {
+        let patch_file_contents = "patch contents";
+
+        let temp_dir = TempDir::new("patch_manager")?;
+        let mut manager = PatchManager::manager_for_test(&temp_dir);
+        assert!(manager.highest_seen_patch_number().is_none());
+
+        // Add patch 1
+        let file_path = &temp_dir.path().join("patch.vmcode");
+        std::fs::write(file_path, patch_file_contents)?;
+        assert!(manager.add_patch(1, file_path).is_ok());
+        assert_eq!(manager.highest_seen_patch_number(), Some(1));
+
+        // Add patch 4, expect 4 to be the highest patch number we've seen
+        let file_path = &temp_dir.path().join("patch.vmcode");
+        std::fs::write(file_path, patch_file_contents)?;
+        assert!(manager.add_patch(4, file_path).is_ok());
+        assert_eq!(manager.highest_seen_patch_number(), Some(4));
+
+        // Add patch 3, expect 4 to still be the highest patch number we've seen
+        let file_path = &temp_dir.path().join("patch.vmcode");
+        std::fs::write(file_path, patch_file_contents)?;
+        assert!(manager.add_patch(3, file_path).is_ok());
+        assert_eq!(manager.highest_seen_patch_number(), Some(4));
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod last_successfully_booted_patch_tests {
+    use super::*;
+    use tempdir::TempDir;
+
+    #[test]
+    fn returns_none_if_no_patch_has_been_booted() -> Result<()> {
+        let temp_dir = TempDir::new("patch_manager").unwrap();
+        let mut manager = PatchManager::manager_for_test(&temp_dir);
+        manager.add_patch_for_test(&temp_dir, 1)?;
+        assert!(manager.last_successfully_booted_patch().is_none());
+
+        Ok(())
+    }
+
+    #[test]
+    fn returns_value_from_patches_state() -> Result<()> {
+        let temp_dir = TempDir::new("patch_manager").unwrap();
+        let mut manager = PatchManager::manager_for_test(&temp_dir);
+        manager.add_patch_for_test(&temp_dir, 1)?;
+
+        let expected = PatchInfo {
+            path: manager.patch_artifact_path(1),
+            number: 1,
+        };
+        manager.patches_state.last_booted_patch = manager.patches_state.next_boot_patch;
+        assert_eq!(manager.last_successfully_booted_patch(), Some(expected));
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod get_next_boot_patch_tests {
+    use super::*;
+    use anyhow::Result;
+    use tempdir::TempDir;
+
+    #[test]
+    fn returns_none_if_no_next_boot_patch() {
+        let temp_dir = TempDir::new("patch_manager").unwrap();
+        let mut manager = PatchManager::with_root_dir(temp_dir.path().to_owned());
+        assert!(manager.next_boot_patch().is_none());
```

Comment:
I'm not sure if this or assert_eq!(value, None) is more idiomatic (you seem to use both in this file).

Reply:
I'm not able to find a clear answer after a couple minutes of googling, but I personally prefer `is_none()` (in the same way I'd prefer `assert!(my_string.is_empty())` to `assert_eq!(my_string, "")`). Just pushed a change to standardize.